application/src/main/webapp/js/chat.js

```diff
@@ -1648,17 +1659,41 @@ ChatApplication.prototype.getUsers = function(roomId, callback, asString) {
  * @param filter : the filter (ex: Ben Pa)
  * @param callback : return the json users data list as a parameter of the callback function
  */
-ChatApplication.prototype.getAllUsers = function(filter, callback) {
+ChatApplication.prototype.getAllUsers = function(filter, isSearchInRoomFirst, callback) {
+
+  // Cache result
+  var _this = this;
+  if (this.getAllUsersCache == undefined) {
+    this.getAllUsersCache = {};
+  }
+  var cacheKey = filter;
+  if (isSearchInRoomFirst) {
+    cacheKey += '__' + this.targetUser;
+  }
+  if (this.getAllUsersCache[cacheKey] != undefined) {
```

Comment:
I agree, we should consider to use a more advanced client-cache if we really need in other issue. 
I will create an issue to take note, ok?